modules/account-lib/src/coin/dot/transactionBuilder.ts

```diff
@@ -1,57 +1,506 @@
+   * @see https://wiki.polkadot.network/docs/build-protocol-info#transaction-mortality
+   */
+  blockHash(blockHash: string): this {
+    this._blockHash = blockHash;
+    return this;
+  }
+  /**
+   *
+   * The current version for transaction format.
+   *
+   * @param {number} transactionVersion
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  version(transactionVersion: number): this {
+    this._transactionVersion = transactionVersion;
+    return this;
+  }
+
+  /**
+   *
+   * The spec name for the registry.
+   *
+   * @param {PolkadotSpecNameType} specName
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  specName(specName: PolkadotSpecNameType): this {
+    this._specName = specName;
+    return this;
+  }
+
+  /**
+   *
+   * The genesis hash of the chain.
+   *
+   * @param {string} genesisHash
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  genesisHash(genesisHash: string): this {
+    this._genesisHash = genesisHash;
+    return this;
+  }
+
+  /**
+   *
+   * The raw metadata of the chain in string format.
+   *
+   * @param {string} metadataRpc
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  metadataRpc(metadataRpc: string): this {
+    this._metadataRpc = metadataRpc;
+    return this;
+  }
+
+  /**
+   *
+   * The specVersion for transaction format.
+   *
+   * @param {number} specVersion
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  specVersion(specVersion: number): this {
+    this._specVersion = specVersion;
+    return this;
+  }
+
+  /**
+   *
+   * The chainName for transaction format.
+   *
+   * @param {number} chainName
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  chainName(chainName: string): this {
+    this._chainName = chainName;
+    return this;
+  }
+
+  private method(method: TxMethod): this {
+    this._method = method;
+    return this;
+  }
+
+  /**
+   * Sets the
+   * specName,
+   * genesisHash - genesisHash of the chain,
+   * metadataRpc - The SCALE-encoded metadata for the runtime when submitted,
+   * specVersion - The current spec version for the runtime,
+   * chainName - chainName,
+   * registry,
+   * of the testnet
+   *
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  testnet(): this {
```

Comment:
This method is also call in TransactionBuilderFactory.ts.
Trying to think how can reuse them without replicating code over there.
@argjv , @gibrangul95 

This applies for testnet(), staticsConfig(), westend(), and mainnet()  methods

Reply:
This was something we thought of as well. I couldnt come up with anything for this given we need this code in both places. Let me know if you have any ideas